test/integration/targets/inventory_constructed/invs/3/keyed_group_str_default_value_with_priority.yml

```
@@ -3,9 +3,11 @@ keyed_groups:
   - key: keyed_group_1
     default_value: "priority_2"
     prefix: b_keyed_group
-    priority: 2
+    vars:
+      ansible_group_priority: 2
 
   - key: keyed_group_2
     default_value: "priority_3"
     prefix: a_keyed_group
-    priority: '{{ 3 if "priority_3" in ansible_keyed_group_name else 1 }}'
+    vars:
+      ansilbe_group_priority: '{{ 3 if "priority_3" in keyed_group_2 else 1 }}'
```

```suggestion
      ansible_group_priority: '{{ 3 if "priority_3" in keyed_group_2 else 1 }}'
```
Also, `Group.set_variable` would also need to be updated to template this value inline- IIRC right now it's just calling `int()`, so when you fix this to the right varname, it's going to bomb now in `set_variable` (or at least ignore the priority set).

---

the templating should happen on the plugin side, specifically for this variable, group object wont have access to all vars

---

my other thought was .. don't make it templatable at all, still debating in my head how useful that would be